src/components/shared/DataTable.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react'
+import Paper from '@mui/material/Paper'
+import Table from '@mui/material/Table'
+import TableBody from '@mui/material/TableBody'
+import TableCell from '@mui/material/TableCell'
+import TableContainer from '@mui/material/TableContainer'
+import TableHead from '@mui/material/TableHead'
+import TablePagination from '@mui/material/TablePagination'
+import TableRow from '@mui/material/TableRow'
+import { Box, FormControlLabel, Switch } from '@mui/material'
+import SearchIcon from '@mui/icons-material/Search'
+import TextField from '@mui/material/TextField'
+import Autocomplete from '@mui/material/Autocomplete'
+import { Search, SearchIconWrapper, StyledInputBase } from '@/utils/searchInputStyles'
+
+interface DataTableProps {
+  searchInput: boolean
+  isSortTextField: boolean
+  sortLabel?: string
+  sortOptionsData?: string[]
+  columnsData: Column[]
+  rowData: any
+}
+
+interface Column {
+  id: string
+  label: string
+  minWidth?: number
+  align?: 'right' | 'left' | 'center' | 'inherit' | 'justify'
+  //   format?: (value: number) => string
+}
+
+export default function DataTable({
+  searchInput,
+  isSortTextField,
+  sortLabel,
+  sortOptionsData,
+  columnsData,
+  rowData
+}: DataTableProps) {
+  const [page, setPage] = React.useState(0)
+  const [rowsPerPage, setRowsPerPage] = React.useState(10)
+  const [dense, setDense] = React.useState(false)
+
+  const columns: readonly Column[] = columnsData
+  const rows = rowData
+
+  let options: any[] = []
+  if (isSortTextField && sortOptionsData) {
+    options = sortOptionsData.map((option) => ({ label: option }))
+  }
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage)
+  }
+
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(+event.target.value)
+    setPage(0)
+  }
+
+  const handleChangeDense = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setDense(event.target.checked)
+  }
+
+  return (
+    <Box>
+      <Paper sx={{ width: '100%', overflow: 'hidden', padding: '0.5em' }}>
+        <Box component="div" sx={{ display: 'flex', gap: '2em', padding: '0.5em 0' }}>
+          {searchInput && (
+            <Search>
+              <SearchIconWrapper>
+                <SearchIcon />
+              </SearchIconWrapper>
+              <StyledInputBase placeholder="Search…" inputProps={{ 'aria-label': 'search' }} />
+            </Search>
```

Comment:
this doesnt seem to handle any state.
also put it into its own component with props about changing the state and `placeholder: string`

Reply:
Fixed